src/Lean/Elab/Tactic/Location.lean

```diff
@@ -39,9 +39,16 @@ def expandOptLocation (stx : Syntax) : Location :=
 
 open Meta
 
-/-- Runs the given `atLocal` and `atTarget` methods on each of the locations selected by the given `loc`.
-If any of the selected tactic applications fail, it will call `failed` with the main goal mvar.
- -/
+/--
+Runs the given `atLocal` and `atTarget` methods on each of the locations selected by the given `loc`.
+
+* If `loc` is a list of locations, runs at each specified hypothesis (and finally the goal if `⊢` is included),
+  and fails if any of the tactic applications fail.
+* If `loc` is `*`, runs at the target first and then the hypotheses in reverse order.
+  If all tactic applications fail, `withLocation` with call `failed` with the main goal mvar.
+  If the tactic application closes the main goal, `withLocation` will then fail with `no goals to be solved`
```

Comment:
Now that you documented it, it looks weird, right? Do users expect this behavior?

Reply:
Agreed that this is weird behavior, it should just close the goal in this case

Reply:
I will separately PR a change in behaviour.

Reply:
#2917